ocp_resources/role.py

```diff
@@ -9,3 +9,62 @@ class Role(NamespacedResource):
     """
 
     api_group = NamespacedResource.ApiGroup.RBAC_AUTHORIZATION_K8S_IO
+
+    def __init__(
+        self,
+        name=None,
+        namespace=None,
+        client=None,
+        api_groups=None,
+        permissions_to_resources=None,
+        verbs=None,
+        teardown=True,
+        yaml_file=None,
+        delete_timeout=TIMEOUT_4MINUTES,
+        **kwargs,
+    ):
+        super().__init__(
+            client=client,
+            namespace=namespace,
+            name=name,
+            teardown=teardown,
+            yaml_file=yaml_file,
+            delete_timeout=delete_timeout,
+            **kwargs,
+        )
+        self.api_groups = api_groups
+        self.permissions_to_resources = permissions_to_resources
+        self.verbs = verbs
+        self.desired_state = {"rules": []}
+
+    def to_dict(self):
+        if not self.res:
+            super().to_dict()
+
+        if not self.yaml_file and self.permissions_to_resources:
+            self.add_rule(
+                api_groups=self.api_groups,
+                permissions_to_resources=self.permissions_to_resources,
+                verbs=self.verbs,
+            )
+
+    def add_rule(
+        self,
+        verbs,
+        api_groups=None,
+        permissions_to_resources=None,
+    ):
+        if not self.res:
+            super().to_dict()
+
+        rule = {"verbs": verbs}
+        if api_groups:
+            rule["apiGroups"] = api_groups
+        if permissions_to_resources:
+            rule["resources"] = permissions_to_resources
+        if rule:
```

Comment:
`if rule:` will always be `True`, you set it in line 60

Reply:
ack